### PR TITLE
eventsPerDay option for Ingestion

### PR DESCRIPTION
**Pull request by vasukalariya**
# Description
Summarize the change.
Added eventsPerDay option for controlling the rate of ingestion.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?
Tested over an interval of 6s for log and metrics data with single and multiple processes. Verified the amount of data ingested and the sleep times.
image
image 

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: tools/sigclient/cmd/sigclient.go
Comment (AndrewHess) [inline tools/sigclient/cmd/sigclient.go:1]: I think `totalEvents` and `eventsPerDay` are mutually exclusive. Can you make sigclient fail if both options are passed?

**Review by AndrewHess — COMMENTED** [review]
Looks pretty good

**Comment by nkunal** [comment]
looks good, once log comments are addressed, it can be merged

File: tools/sigclient/cmd/sigclient.go
Comment (vasukalariya) [reply tools/sigclient/cmd/sigclient.go:1]: Actually we are ignoring the totalEvents in the case of continuous. So, do we want these manual checks or bypass silently. Cause if we do it for eventsPerDay we should do it for continuous too.

Comment (nkunal) [reply tools/sigclient/cmd/sigclient.go:1]: technically you are right Vasu, but just for code clarity (and readme description) , we can have explicit exit, if both options are given

File: tools/sigclient/pkg/ingest/ingest.go [hunk 1]
```diff
@@ -289,6 +294,14 @@ func runIngestion(iType IngestType, rdr utils.Generator, wg *sync.WaitGroup, url
...
 		atomic.AddUint64(ctr, uint64(recsInBatch))
+		if estimatedMilliSecsPerBatch > 0 {
+			timeTaken := int(time.Since(startTime).Milliseconds())
+			if timeTaken < estimatedMilliSecsPerBatch {
+				napTime := estimatedMilliSecsPerBatch - timeTaken
+				log.Infof("ProcessId: %v finished early in %v ms. Sleeping for %+v ms before next batch", processNo, timeTaken, napTime)
+				time.Sleep(time.Duration(napTime) * time.Millisecond)
+			}
+		}
 	}
```
[the remark this answers is not in the document]
Comment (vasukalariya) [reply tools/sigclient/pkg/ingest/ingest.go]: addressed in the latest_commit.

**Review by AndrewHess — DISMISSED** [review]

File: tools/sigclient/cmd/sigclient.go
Comment (vasukalariya) [reply tools/sigclient/cmd/sigclient.go:1]: addressed in the latest commit.

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**